source/Detector/Scaler/AMDSScalerDetector.cpp

```diff
@@ -265,11 +269,79 @@ void AMDSScalerDetector::onTriggerDwellTimerTimeout()
 {
 	if(triggerDwellInterfaceDwellModeControl_->withinTolerance(0.0)){
 		triggerDwellTimer_->stop();
-		triggerDwellInterfaceStartControl_->move(0);
 	}
 	emit requestFlattenedData(triggerDwellInterfaceDwellTimeControl_->value());
 }
 
+void AMDSScalerDetector::startScalerDwellAcquisition()
+{
+	double seconds = triggerDwellInterfaceDwellTimeControl_->value();
+	if(seconds > 0 && seconds < 100) {
+		if (triggerDwellInterfaceDwellModeControl_->withinTolerance(0.0)) {
+			// set the mapping of value changed. NOTE: This should be in a more graceful way
+			waitingChannels_.clear();
+
+			QMap<quint8, int> channelToIndexMap;
+			for(int x = 0, size = scalerConfiguration_->configuredChannels().count(); x < size; x++)
+				channelToIndexMap.insert(scalerConfiguration_->configuredChannels().at(x), x);
+
+
+			for(int x = 0, size = triggerDwellInterfaceChannelFeedbackControls_.count(); x < size; x++){
+				QString controlPVName = triggerDwellInterfaceChannelFeedbackControls_.value(x)->readPVName();
+
+				QMap<quint8, int>::const_iterator i = channelToIndexMap.constBegin();
+				while(i != channelToIndexMap.constEnd()){
+					QString channelIdTemplate = i.key() < 10 ? QString("0%1").arg(i.key()) : QString("%1").arg(i.key()); // when channel id is 0 ~ 9, the PV is "0" + channelId
+					if(controlPVName.remove("BL1611-ID-1:AMDS:scaler").contains(channelIdTemplate)) {
+						waitingChannels_.append(x);
+						channelValueChangedSignalMapper_->setMapping(triggerDwellInterfaceChannelFeedbackControls_.value(x), x);
+					}
+
+					i++;
+				}
+			}
+			connect(channelValueChangedSignalMapper_, SIGNAL(mapped(int)), this, SLOT(onScalerChannelValueChanged(int)));
+		}
+
+		// start the time
+		int asMsecs = (int)(seconds*1000);
+		triggerDwellTimer_->setInterval(asMsecs);
+		triggerDwellInterfaceDwellStateControl_->move(1.0);
+		triggerDwellTimer_->start();
+	}
+}
+
+void AMDSScalerDetector::stopScalerDwellAcquisition(bool forceStop)
+{
+	if (forceStop) {
+		if (triggerDwellTimer_->isActive())
+			triggerDwellTimer_->stop();
+
+		foreach(int channelIndex, waitingChannels_) {
+			waitingChannels_.removeAll(channelIndex);
+			channelValueChangedSignalMapper_->removeMappings(triggerDwellInterfaceChannelFeedbackControls_.value(channelIndex));
+		}
+	}
+
+	if(waitingChannels_.count() == 0) {
+		disconnect(channelValueChangedSignalMapper_, SIGNAL(mapped(int)), this, SLOT(onScalerChannelValueChanged(int)));
+
+		triggerDwellInterfaceStartControl_->move(0);
+		triggerDwellInterfaceDwellStateControl_->move(0);
+	}
+}
+
+void AMDSScalerDetector::onScalerChannelValueChanged(int channelIndex)
+{
+	if(waitingChannels_.contains(channelIndex)){
```

Comment:
Does this function only remove channels?  If so, I think this function should be renamed.